File: src/main/java/com/goods/partner/entity/Route.java
package com.goods.partner.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.time.LocalDateTime;

@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "routes")
public class Route {

    @Id
    private int id;

    private String status;
    private double totalWeight;
    private int totalPoints;
    private int totalOrders;
    private double distance;
    private LocalDateTime estimatedTime;
    private LocalDateTime startTime;
    private LocalDateTime finishTime;
    private LocalDateTime spentTime;
    private String routeLink;
    private String storeName;
    private String storeAddress;
}

File: goods-partner-mobile/model/Route.ts
import MapPoint from './MapPoint'
import {RoutePoint} from './RoutePoint'
import {Car} from './Car'

export interface Route {
    car: Car
    distance: number
    estimatedTime: number
    finishTime: Date
    id: number
    optimization: boolean
    routePoints: RoutePoint[]
    spentTime: number
    startTime: Date
    status: RouteStatus
    store: Store
    totalOrders: number
    totalPoints: number
    totalWeight: number
}

export interface Store {
    address: string
    mapPoint: MapPoint
    name: string
}

export enum RouteStatus {
    DRAFT = "DRAFT",
    APPROVED = "APPROVED",
    INPROGRESS = "INPROGRESS",
    COMPLETED = "COMPLETED"

}
